contracts/Collector.sol

```diff
@@ -52,7 +70,7 @@ contract Collector is ICollector{
     onlyOwner()
     {
         uint balance = token.balanceOf(address(this));
-        require(balance > 0, "no revenue to share");
+        require(balance > partners.length, "no revenue to share");
 
         for(uint i = 0; i < partners.length; i++)
             token.transfer(partners[i].beneficiary, SafeMath.div(SafeMath.mul(balance, partners[i].share), 100));
```

Comment:
we should not do this. Each partner should trigger their own withdrawal. the contract should not send value.
See https://docs.soliditylang.org/en/v0.8.15/common-patterns.html#withdrawal-from-contracts

Reply:
I think that we should create a specific task for this.

Reply:
Created a [task](https://rsklabs.atlassian.net/browse/PP-351) for this.